qp/ensemble.py

```diff
@@ -201,6 +214,59 @@ def objdata(self):
         dd.update(self._gen_obj.objdata)
         return dd
 
+    def set_ancil(self, ancil):
+        """Set the ancillary data dict
+
+        Parameters
+        ----------
+        ancil : `dict`
+            The ancillary data dictionary
+
+        Notes
+        -----
+        Raises IndexError if the length of the arrays in ancil does not match
+        the number of PDFs in the Ensemble
+        """
+        check_array_shapes(ancil, self.npdf)
+        self._ancil = ancil
+
+    def add_to_ancil(self, to_add):  # pragma: no cover
+        """ Add additionaly columns to the ancillary data dict
+
+        Parameters
+        ----------
+        to_add : `dict`
+            The columns to add to the ancillary data dict
+
+        Notes
+        -----
+        Raises IndexError if the length of the arrays in to_add does not match
+        the number of PDFs in the Ensemble
+
+        This calls dict.update() so it will overwrite existing columns
+        """
+        check_array_shapes(to_add, self.npdf)
+        self._ancil.update(to_add)
+
+
+    def append(self, other_ens):
+        """ Append another other_ens to this one
+
+        Parameters
+        ----------
+        other_ens : `qp.Ensemble`
+            The other Ensemble
+        """
+        if not compare_dicts([self.metadata(), other_ens.metadata()]):  # pragma: no cover
```

Comment:
There seems to be a bug here when trying to append two e.g. qp.interp ensembles that have an array as one of the dictionary keys (xvals stores the redshift grid as an array) that raises `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

So, I think we need to go through the keys and check for arrays and add a .all for those.  It seems like doing a check_keys rather than compare_dicts might work if check_keys returned a true when the dictionaries matched.  Actually, I think this same issue cropped up somewhere else so this may already be implemented somewhere.